Add Step Function Visualization Documentation (#20468)
DataDog/documentation: content/en/serverless/step_functions/_index.md

AWS Step Functions is a serverless orchestration service that lets developers create and manage multi-step application workflows. In addition to getting Cloudwatch metrics from Datadog's [AWS Step Functions integration][2], Datadog also provides native AWS Step Function monitoring and tracing.

{{< img src="serverless/step_functions/overview1.png" alt="An AWS Step Function oveview tab." style="width:100%;" >}}

### Monitor the overall health of Step Functions in the Serverless view
The Serverless view shows key metrics for your Step Functions in one place to easily provide a snapshot of the health of your Step Functions. You can access a detailed view of each Step Function to see all associated metrics, logs, and traces within a certain time frame and set monitors for problematic executions. 

{{< img src="serverless/step_functions/overview2.png" alt="An AWS Step Function visualization with span tags." style="width:100%;" >}}

###  Visualize AWS Step Function Traces on a State Machine Map

> When [Step Function tracing is enabled][1], you can use a visual representation of an AWS Step Function execution through a state machine map. Get an at-a-glance-view of whether a successful or failed execution took the expected path through the state machine without having to follow a full flame graph. Drill into any anomalous executions to identify what states were problematic or had a high latency with ease.

Reviewer: Optional
```suggestion
When [Step Function tracing is enabled][1], you can use a visual representation of an AWS Step Function execution through a state machine map. Get an at-a-glance-view of whether a successful or failed execution took the expected path through the state machine. Drill into any anomalous executions to identify what states are problematic or have a high latency.
```
